### PR TITLE
Erin King

### PR DESCRIPTION
**Pull request by erin-king**

### CODE REVIEW

File: lib/market.rb [hunk 1]
```diff
@@ -1,3 +1,62 @@
...
+    @name = name
+    @vendors = []
+  end
+
+  def add_vendor(vendor)
+    @vendors << vendor
+  end
+
+  def vendor_names
+    @vendors.map do |vendor|
```
Comment (ameseee) [inline lib/market.rb]: Whoohoo🎉

File: lib/market.rb [hunk 2]
```diff
@@ -1,3 +1,62 @@
...
+      vendor.name
+    end
+  end
+
+  def vendors_that_sell(item)
+    @vendors.find_all do |vendor|
+      vendor.check_stock(item) > 0
```
Comment (ameseee) [inline lib/market.rb]: beautiful! another option is `vendor.inventory.includes? (vendor)` - just throwing it out there, maybe you also considered that though!

File: lib/market.rb [hunk 3]
```diff
@@ -1,3 +1,62 @@
...
+    end
+  end
+
+  def sorted_item_list
+    item_list = []
+    @vendors.each do |vendor|
+      item_list << vendor.inventory.keys
+    end
+    item_list.flatten.uniq.sort
```
Comment (ameseee) [inline lib/market.rb]: Do you really need to flatten?

File: lib/market.rb [hunk 4]
```diff
@@ -1,3 +1,62 @@
...
+  end
+
+  def total_inventory
+    total_inventory_hash = Hash.new(0)
```
Comment (ameseee) [inline lib/market.rb]: nice job using `Hash.new(0)` - curious why you didn't use `reduce/inject`? Not a huge deal, just to think about.

File: lib/market.rb [hunk 5]
```diff
@@ -1,3 +1,62 @@
...
+    @vendors.each do |vendor|
+      vendor.inventory.each do |item, amount|
+        total_inventory_hash[item] += amount
+      end
+    end
+    total_inventory_hash
+  end
+
+  def sell(item, amount)
```
Comment (ameseee) [inline lib/market.rb]: nice work! Would love to push you to find a way to refactor this so no method is longer than 8 lines💪💪💪

File: lib/vendor.rb [hunk 6]
```diff
@@ -4,7 +4,15 @@ class Vendor
 
   def initialize(name)
     @name = name
-    @inventory = # you decide what goes here...
+    @inventory = Hash.new(0)
```
Comment (ameseee) [inline lib/vendor.rb]: 💯

File: test/market_test.rb [hunk 7]
```diff
@@ -5,4 +5,105 @@
...
+
+  def test_it_has_attributes
+    assert_equal "South Pearl Street Farmers Market", @market.name
+  end
+
+  def test_it_starts_with_no_vendors
+    assert_equal [], @market.vendors
+  end
+
+  def test_it_can_add_vendors
```
Comment (ameseee) [inline test/market_test.rb]: Would love to see a bit of an edge-case test here, checking that you can't add the same vendor more than once😉

**Comment by ameseee** [comment]
@erin-king you are crushing it. A few comments for your review - really just trying to push you to go above and beyond, but want it to be clear that you are doing really great in terms of Mod1 learning goals.
